bootstrap/app.go

```diff
@@ -102,6 +103,13 @@ func (a *app) Run() {
 				a.logger.Panicf("%v", runErr)
 			}
 		}()
+		if a.config.Prometheus.Enable {
+			go func() {
+				if runErr := prometheus.Run(a.config.Prometheus.ExportPath, a.config.Prometheus.Port); runErr != nil {
+					a.logger.Panicf("%v", runErr)
+				}
+			}()
+		}
```

Comment:
Shall we move the `prometheus` instance under `IndexService`?

Instead of directly importing `prometheus` to `bootstrap/syncmanager.go` and `entity/projection/manager.go`, I think we could do it through dependency injection.

Reply:
I though about this as well, but that will mean anywhere we want to use `prometheus` will need many extra param passing. Let say in views, projections, parser or even outer package logic.

Or any example we will need DI for prometheus?

Reply:
There are a few reasons why I prefer to use dependency injection.

1. Readability. If not do it through dependency injection, it might confuse future developers. As they may spend more time to understand where does it initialize and how to use it.

2. Convention. I think `prometheus` is kind of like the existing logger. We already apply dependency injection for logger, maybe we should also do that for `prometheus`?

3. Unit Test. If doing through dependency injection, it will be easier for writing unit test. As we could make use of `interface`. If not do through dependency injection, it may be tricky to write unit test for it.

Reply:
Understood, i will change to a package dependency injection.

Reply:
Updated for a package dependency injection only. Since prometheus still better to keep it as an external package in case we can use in any parts of the app. Make it instance would result in unnecessary code complexity.  

For example, we would use it in the indexService, httpAPIServer, projections, views, parsers, or others external package.